src/browser/browser.ts

```diff
@@ -575,6 +604,22 @@ export class Browser {
     }
   }
 
+  removeTracingHeader = (req: puppeteer.HTTPRequest) => {
```

Comment:
Wouldn't be safe to try/catch this code? I'm mostly concerned about `new URL(referer)` if it is not a URL it will fail but in general, it sounds safer to try/catch the entire block and take a decision if it fails (maybe just continue without deleting the headers